db/migrate/20190611144112_create_notification_subscriptions.rb

```diff
@@ -0,0 +1,12 @@
+class CreateNotificationSubscriptions < ActiveRecord::Migration[5.2]
+  def change
```

Comment:
I would add an index in this table. You select these two options:

`NotificationSubscription.where(notifiable_id: comment.commentable_id, notifiable_type: "Article", config: "all_comments")`

and 

`NotificationSubscription.where(notifiable_id: comment.commentable_id, notifiable_type: "Article", config: "top_level_comments")`

I think it would benefit from a composite index on `notifiable_id`, `notifiable_type` and `config`. 

Something like `t.index [:notifiable_id, :notifiable_type, :config]` I guess